tenseal/cpp/tensors/bfvvector.cpp

```diff
@@ -282,40 +345,51 @@ shared_ptr<BFVVector> BFVVector::enc_matmul_plain_inplace(
 }
 
 shared_ptr<BFVVector> BFVVector::replicate_first_slot_inplace(size_t n) {
+    if (this->_ciphertexts.size() != 1)
+        throw invalid_argument("can't replicate first_slot on chunked vectors");
+
     // mask
-    vector<int64_t> mask(this->_size, 0);
+    vector<int64_t> mask(this->size(), 0);
     mask[0] = 1;
     this->mul_plain_inplace(mask);
 
     // replicate
-    Ciphertext tmp = this->_ciphertext;
+    Ciphertext tmp = this->_ciphertexts[0];
     auto galois_keys = this->tenseal_context()->galois_keys();
     for (size_t i = 0; i < (size_t)ceil(log2(n)); i++) {
         this->tenseal_context()->evaluator->rotate_vector_inplace(
             tmp, static_cast<int>(-pow(2, i)), *galois_keys);
-        this->tenseal_context()->evaluator->add_inplace(this->_ciphertext, tmp);
-        tmp = this->_ciphertext;
+        this->tenseal_context()->evaluator->add_inplace(this->_ciphertexts[0],
+                                                        tmp);
+        tmp = this->_ciphertexts[0];
     }
 
-    this->_size = n;
+    this->_sizes = {n};
     return shared_from_this();
 }
```

Comment:
Can't we implement this on multi-ciphertext vectors? I was thinking of two possibilities:
1. n is bigger than available slots, then we can just fill the first ct with the first value, then replicate that ciphertext until having enough cts.
2. n is smaller than available slots, then we can manipulate the first ct as we are doing right now